cheerio-scraper/src/crawler_setup.js

```diff
@@ -244,10 +233,12 @@ class CrawlerSetup {
      *
      * Finally, it makes decisions based on the current state and post-processes
      * the data returned from the `pageFunction`.
-     * @param {Object} environment
+     * @param {Object} crawlingContext
      * @returns {Function}
      */
-    async _handlePageFunction({ request, response, $, body, json, contentType, autoscaledPool }) {
+    async _handlePageFunction(crawlingContext) {
+        const { request, response, $, body, contentType, crawler } = crawlingContext;
```

Comment:
There's a potential for a performance improvement in not destructuring the `body` because by doing that, you call the getter in `CheerioCrawler` and force serialization of the whole request body, essentially doubling the memory consumption.

This could have huge impact on high-performance scrapers which are strapped for resources. We actually introduced the streaming parser to resolve memory leaks. Before that, actors were often memory constrained instead of CPU constrained.

I remember that I decided to not solve this problem earlier, because the property is also reassigned in `Context` which would also call the getter. But maybe now there is a better way.

Reply:
`body` is no longer destructed as it was needed only in the `html` getter that is now gone. Looked at the `Context` implementation and that uses this to get around getter evaluation, so it should be ok I guess?

```ts
        // Page function arguments are directly passed from CrawlerSetup
        // and differ between Puppeteer and Cheerio Scrapers.
        // We must use properties and descriptors not to trigger getters / setters.
        Object.defineProperties(this, Object.getOwnPropertyDescriptors(pageFunctionArguments));
```

Reply:
Ok, cool. We'll see. It will definitely not be worse than before :)